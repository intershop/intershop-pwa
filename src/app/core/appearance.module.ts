import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
