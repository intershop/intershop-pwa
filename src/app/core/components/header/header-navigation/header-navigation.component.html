<ul class="navbar-nav main-navigation-list accordion d-md-none" [class.accordion]="view!=='full'">
  <ng-container *ngTemplateOutlet="categoryTemplate"></ng-container>
</ul>
<ul class="navbar-nav main-navigation-list d-none d-md-block" [class.accordion]="view==='small'">
  <ng-container *ngTemplateOutlet="categoryTemplate"></ng-container>
</ul>
<!--- Category Template -->
<ng-template #categoryTemplate>
  <li *ngFor="let category of categories" class="dropdown" [ngClass]="{'open' : isOpened(category)}" #submenu (mouseover)="subMenuShow(submenu)"
    (mouseout)="subMenuHide(submenu)" (click)="subMenuHide(submenu)">
    <a [routerLink]="['/category', category.uniqueId]" [attr.data-testing-id]="category.uniqueId + '-link'" [ngStyle]="{'width' : (!category.hasChildren()) ? '100%' : ''}">
      {{category.name}}
    </a>
    <a *ngIf="category.hasChildren()" class="dropdown-toggle d-md-none" [class.d-md-none]="view==='auto'" [class.d-none]="view=='full'"
      (click)="toggleOpen(category)">
      <fa-icon *ngIf="isOpened(category); else closed" [icon]="['fas', 'minus']"></fa-icon>
      <ng-template #closed>
        <fa-icon [icon]="['fas', 'plus']"></fa-icon>
      </ng-template>
    </a>
    <ish-sub-category-navigation [category]="category" [subCategoriesDepth]="1" [view]="view"></ish-sub-category-navigation>
  </li>
</ng-template>
