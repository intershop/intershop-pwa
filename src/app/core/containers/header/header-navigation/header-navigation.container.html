<ish-header-navigation [categories]="categories$ | async" [view]="view"></ish-header-navigation>
