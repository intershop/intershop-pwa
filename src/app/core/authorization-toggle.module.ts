import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
