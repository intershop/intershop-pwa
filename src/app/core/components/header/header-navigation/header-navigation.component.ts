import { Component, Input, OnInit } from '@angular/core';
import { Category } from '../../../../models/category.model';
import { CategoriesService } from '../../../../shared/services/categories/categories.service';
import { CurrentLocaleService } from '../../../services/locale/current-locale.service';

@Component({
  selector: 'ish-header-navigation',
  templateUrl: './header-navigation.component.html'
})

export class HeaderNavigationComponent implements OnInit {

  @Input() maxSubCategoriesDepth = 0;
  categories: Category[];

  constructor(
    public categoriesService: CategoriesService,
    private currentLocaleService: CurrentLocaleService
  ) { }

  ngOnInit() {
    this.currentLocaleService.subscribe(() => {
      this.categoriesService.getTopLevelCategories(this.maxSubCategoriesDepth).subscribe((response: Category[]) => {
        if (typeof (response) === 'object') {
          this.categories = response;
        }
      });
    });
  }
}
