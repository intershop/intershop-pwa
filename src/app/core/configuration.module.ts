import { NgModule } from '@angular/core';

import { SPECIAL_HTTP_ERROR_HANDLER } from './interceptors/icm-error-mapper.interceptor';
import { createPaymentErrorHandler } from './utils/http-error/create-payment.error-handler';
import { editPasswordErrorHandler } from './utils/http-error/edit-password.error-handler';
import { LoginUserErrorHandler } from './utils/http-error/login-user.error-handler';
import { requestReminderErrorHandler } from './utils/http-error/request-reminder.error-handler';
import { updatePasswordErrorHandler } from './utils/http-error/update-password.error-handler';

@NgModule({
  providers: [
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: updatePasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useClass: LoginUserErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: requestReminderErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: editPasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: createPaymentErrorHandler, multi: true },
  ],
})
export class ConfigurationModule {}
