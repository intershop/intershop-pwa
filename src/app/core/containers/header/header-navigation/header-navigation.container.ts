import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';
import { Category } from '../../../../models/category/category.model';
import { getTopLevelCategories } from '../../../../shopping/store/categories';
import { CoreState } from '../../../store/core.state';

@Component({
  selector: 'ish-header-navigation-container',
  templateUrl: './header-navigation.container.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class HeaderNavigationContainerComponent implements OnInit {

  categories$: Observable<Category[]>;

  constructor(
    private store: Store<CoreState>,
  ) { }

  ngOnInit() {
    this.categories$ = this.store.pipe(select(getTopLevelCategories));
  }
}
