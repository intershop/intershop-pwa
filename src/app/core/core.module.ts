import { APP_BASE_HREF, PlatformLocation } from '@angular/common';
import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { ErrorHandler, NgModule, Optional, SkipSelf } from '@angular/core';
import { ServiceWorkerModule } from '@angular/service-worker';
import { TranslateModule } from '@ngx-translate/core';

import { AppearanceModule } from './appearance.module';
import { ConfigurationModule } from './configuration.module';
import { FeatureToggleModule } from './feature-toggle.module';
import { IdentityProviderModule } from './identity-provider.module';
import { ICMErrorMapperInterceptor } from './interceptors/icm-error-mapper.interceptor';
import { IdentityProviderInterceptor } from './interceptors/identity-provider.interceptor';
import { MockInterceptor } from './interceptors/mock.interceptor';
import { InternationalizationModule } from './internationalization.module';
import { StateManagementModule } from './state-management.module';
import { DefaultErrorHandler } from './utils/default-error-handler';

@NgModule({
  imports: [
    AppearanceModule,
    ConfigurationModule,
    FeatureToggleModule,
    HttpClientModule,
    IdentityProviderModule,
    InternationalizationModule,
    ServiceWorkerModule.register('ngsw-worker.js', { enabled: SERVICE_WORKER }),
    StateManagementModule,
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: ICMErrorMapperInterceptor, multi: true },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: IdentityProviderInterceptor,
      multi: true,
    },
    { provide: HTTP_INTERCEPTORS, useClass: MockInterceptor, multi: true },
    { provide: ErrorHandler, useClass: DefaultErrorHandler },
    {
      provide: APP_BASE_HREF,
      useFactory: (s: PlatformLocation, baseHref: string) => baseHref || s.getBaseHrefFromDOM(),
      deps: [PlatformLocation, [new Optional(), new SkipSelf(), APP_BASE_HREF]],
    },
  ],
  // exports needed to use the cookie banner in the AppComponent
  exports: [TranslateModule],
})
export class CoreModule {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error('CoreModule is already loaded. Import it in the AppModule only');
    }
  }
}
