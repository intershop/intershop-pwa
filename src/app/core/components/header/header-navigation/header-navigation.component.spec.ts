import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { HeaderNavigationComponent } from './header-navigation.component';

describe('Header Navigation Component', () => {
  let fixture: ComponentFixture<HeaderNavigationComponent>;
  let component: HeaderNavigationComponent;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        HeaderNavigationComponent
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderNavigationComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(function() { fixture.detectChanges(); }).not.toThrow();
  });
});
