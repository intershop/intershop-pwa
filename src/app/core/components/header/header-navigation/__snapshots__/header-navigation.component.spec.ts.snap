// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Navigation Component should be created 1`] = `
<div
  id="root0"
  ng-version="6.1.6"
>
  <ul
    class="navbar-nav main-navigation-list d-md-none accordion"
    ng-reflect-klass="navbar-nav main-navigation-lis"
    ng-reflect-ng-class="[object Object]"
  >
    
    
    <li
      class="dropdown"
      ng-reflect-klass="dropdown"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        data-testing-id="A-link"
        ng-reflect-ng-style="[object Object]"
        style="width: 100%;"
      >
         CAT_A 
      </a>
      
      <ish-sub-category-navigation />
    </li>
    <li
      class="dropdown"
      ng-reflect-klass="dropdown"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        data-testing-id="B-link"
        ng-reflect-ng-style="[object Object]"
        style="width: 100%;"
      >
         CAT_B 
      </a>
      
      <ish-sub-category-navigation />
    </li>
    <li
      class="dropdown"
      ng-reflect-klass="dropdown"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        data-testing-id="C-link"
        ng-reflect-ng-style="[object Object]"
        style="width: 100%;"
      >
         CAT_C 
      </a>
      
      <ish-sub-category-navigation />
    </li>
    
  </ul>
  <ul
    class="navbar-nav main-navigation-list d-none d-md-block"
    ng-reflect-klass="navbar-nav main-navigation-lis"
    ng-reflect-ng-class="[object Object]"
  >
    
    
    <li
      class="dropdown"
      ng-reflect-klass="dropdown"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        data-testing-id="A-link"
        ng-reflect-ng-style="[object Object]"
        style="width: 100%;"
      >
         CAT_A 
      </a>
      
      <ish-sub-category-navigation />
    </li>
    <li
      class="dropdown"
      ng-reflect-klass="dropdown"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        data-testing-id="B-link"
        ng-reflect-ng-style="[object Object]"
        style="width: 100%;"
      >
         CAT_B 
      </a>
      
      <ish-sub-category-navigation />
    </li>
    <li
      class="dropdown"
      ng-reflect-klass="dropdown"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        data-testing-id="C-link"
        ng-reflect-ng-style="[object Object]"
        style="width: 100%;"
      >
         CAT_C 
      </a>
      
      <ish-sub-category-navigation />
    </li>
    
  </ul>
  
</div>
`;
